pennylane/data/base/_hdf5.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains a lazy-loaded interface to the HDF5 module. For internal use only."""
+
+import importlib
+from pathlib import Path
+from types import ModuleType
+from typing import Any, Literal, Union, Optional
+from uuid import uuid4
+from functools import lru_cache
+
+
+from .typing_util import HDF5Any, HDF5Group
+
+_MISSING_MODULES_EXC = ImportError(
+    "This feature requires the 'aiohttp', 'h5py' and 'fsspec' packages. "
+    "They can be installed with:\n\n pip install aiohttp fsspec h5py"
```

Comment:
just realized after spotting the merge-conflict (Rashid just totally re-did our CI workflows and they are much nicer 😄 ) that our tests only pip install `h5py`. Does something make a difference here vs tests?

Reply:
Ah right, `fsspec` and `aiohttp` are only needed for loading datasets from S3. So we shouldn't need it here